src/cheat_gui.cpp

```diff
@@ -266,16 +266,17 @@ struct CheatWindow : Window {
 
 	void DrawCheatWidget(const Rect &r) const
 	{
+		const Dimension setting_button = GetSettingButtonSize();
 		const Rect ir = r;
 		int y = ir.top;
 
 		bool rtl = _current_text_dir == TD_RTL;
-		uint button_left = rtl ? ir.right - SETTING_BUTTON_WIDTH : ir.left;
-		uint text_left   = ir.left + (rtl ? 0 : WidgetDimensions::scaled.hsep_wide + SETTING_BUTTON_WIDTH);
-		uint text_right  = ir.right - (rtl ? WidgetDimensions::scaled.hsep_wide + SETTING_BUTTON_WIDTH : 0);
+		uint button_left = rtl ? ir.right - setting_button.width : ir.left;
+		uint text_left   = ir.left + (rtl ? 0 : WidgetDimensions::scaled.hsep_wide + setting_button.width);
+		uint text_right  = ir.right - (rtl ? WidgetDimensions::scaled.hsep_wide + setting_button.width : 0);
 
 		int text_y_offset = (this->line_height - GetCharacterHeight(FS_NORMAL)) / 2;
-		int button_y_offset = (this->line_height - SETTING_BUTTON_HEIGHT) / 2;
+		int button_y_offset = (this->line_height - setting_button.height) / 2;
```

Comment:
Cases like this:
`line_height` is unsigned. Previously the signed `SETTING_BUTTON_HEIGHT` forced a promotion to signed. Now it is unsigned.
If for some reason `line_height` is smaller, `button_y_offset` will end up as huge positive, instead of small negative.

In this case `line_height` seems to include the button height. But checking all cases...